autorest/azure/example/main.go

```diff
@@ -244,7 +244,7 @@ func main() {
 	}
 
 	client := &autorest.Client{}
-	client.Authorizer = spt
+	client.Authorizer = autorest.NewBearerAuthorizer(spt)
```

Comment:
Is it not possible to make the `spt` type implement the authorizer interface so that it's slightly more backward compatible? Or is that less clean?